frontend/src/services/http/http.ts

```diff
@@ -0,0 +1,103 @@
+import { api } from '../../constants/api';
+import { HttpError } from '../../util/HttpError';
+
+type method = 'GET' | 'POST' | 'PUT' | 'DELETE';
+
+/**
+ * The HTTP service provides standardized configurations for making requests.
+ *
+ * The passed body is stringified if such is provided and all necessary headers are
+ * attached
+ *
+ * If the server responds with 401, the service will attempt to retrieve
+ * a new access token and then redo the request.
+ *
+ * If the server responds with 204, the service returns ``undefined``. For
+ * other 2xx status codes, it returns the response body. For any
+ * other response status code, an ``HttpError`` with the response is thrown.
+ */
+export const httpService = {
+  get<TResponseBody>(url: string) {
+    return request<TResponseBody>('GET', url);
+  },
+  post<TResponseBody>(url: string, body?: any) {
+    return request<TResponseBody>('POST', url, body);
+  },
+  put<TResponseBody>(url: string, body?: any) {
+    return request<TResponseBody>('PUT', url, body);
+  },
```

Comment:
We also need a `patch` method.

Reply:
PATCH has been added to the HTTP service